libraries/splunk_password.rb

```diff
@@ -0,0 +1,52 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# File Name:: splunk_password.rb
+
+require 'base64'
+
+# Module contains different functions to encrypt and decrypt splunk passwords
+module CernerSplunk
+  # Encrypts the password before writing into config files. As of now all the passwords
+  # needs to be XORed except for the sslKeyFilePassword. The boolean
+  # parameter xor controls the XOR logic.
+  def self.splunk_encrypt_password(plain_text, splunk_secret, xor = true)
+    fail 'string to be encrypted cannot be empty/nil' if plain_text.nil? || plain_text.empty?
+    rc4key = splunk_secret.strip[0..15]
+
+    if !xor
+      pwd = plain_text + "\0"
+    else
+      pwd = plain_text.unpack('c*')
+      xorkey = get_xor_key(pwd.size)
+      pwd = pwd.zip(xorkey).map { |c1, c2| c1 ^ c2 }.pack('c*') + "\0"
+    end
+
+    pwd = CernerSplunk::RC4.new(rc4key).encrypt(pwd)
+    '$1$' + Base64.encode64(pwd).strip!
+  end
+
+  # Decrypts the splunk passwords. As of now the encrypted passwords needs to be XORed
+  # to retrieve the plain_text for every password except the sslKeyFilePassword.
+  # The boolean parameter xor controls the XOR logic.
+  def self.splunk_decrypt_password(encryp_password, splunk_secret, xor = true)
+    fail 'string to be decrypted cannot be empty/nil' if encryp_password.nil? || encryp_password.empty?
+
+    rc4key = splunk_secret.strip[0..15]
+    pwd = CernerSplunk::RC4.new(rc4key).decrypt(Base64.decode64(encryp_password.gsub('$1$', '')))
```

Comment:
+1 on making decrypt a more exact reversal of encrypt. Also on the line below it could be .chomp("\0") rather than .delete() to only strip that from the end of the password.